scripts/prep_release.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -euo pipefail
+
+# prep_release.sh - prepares release candidate for testing
+#
+# This script is written to be excessively modular so it can be debugged or restarted easily.
+#
+# Edward Smith, December 2016
+
+scriptfile="$( cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+scriptfile="${scriptfile}"/$(basename "$0")
+cd $(dirname "$scriptfile")/..
+
+scriptFailed=1
+function finish {
+    if [ $scriptFailed -ne 0 ]; then
+        echo ">>> Error: `basename "$scriptfile"` failed!" 1>&2
+        exit 1
+    fi
+}
+trap finish EXIT
+
+
+version=$(./scripts/version.sh)
+
+echo ""
+echo "Before continuing:"
+echo "- Make sure that the release version number is already updated."
+echo "- Make sure that the ChangeLog.md has been updated, spell checked, and is coherent."
+echo "- Unit tests pass in Xcode"
+echo ""
+if ! ./scripts/askYN.sh "Build Branch release candidate version ${version}?"; then
+    echo ">>> Nothing built." 1>&2
+    exit 1
+fi
+
+# Prompt for editor input for ChangeLog.
+nano ChangeLog.md
```

Comment:
One thing you might consider here is `$VISUAL ChangeLog.md` or `${VISUAL:nano} ChangeLog.md`, in case anyone else ever has to cut a release. I'm not familiar with nano and had to find my way out when trying this script. I think `VISUAL` is still valid, or maybe `EDITOR` to allow a user to override with `VISUAL=my-fave-editor scripts/prepare_release.sh`. But not a big deal if that doesn't seem to work well.

Reply:
Good call, i'll make that change.